src/line/funcs/testDataSent.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @jest-environment jsdom
+ */
+const fs = require('fs');
+require('@testing-library/jest-dom');
+const domTesting = require('@testing-library/dom');
+const userEvent = require('@testing-library/user-event').default;
+const functions = require('./functions.js');
+
+const [htmlPath, jsPath] = functions.getPaths();
+
+async function testDataSent(elements, title, X_input, Y_input, X_label, 
+                            Y_label){
+
+    // init dom elements but dont use jest.isolateModules as noted by alex
+    functions.initDomFromFiles(htmlPath, jsPath);
+    functions.initElements(elements);
+ 
+    // mocked generateChartImg function - return valid img URL
+    const gen_chart_spy = require('../../lib/generateChartImg.js');
+    jest.mock('../../lib/generateChartImg.js');
+    gen_chart_spy.mockImplementation(() => { return 
+                                        'http://placekitten.com/480/480'; });
+
+    // do action: type in information to generate our chart with
+    const user = userEvent.setup();
+    await user.type(elements.title_input, title);
+    await user.type(elements.X_inputs[0], X_input);
+    await user.type(elements.Y_inputs[0], Y_input);
+    await user.type(elements.X_label, X_label);
+    await user.type(elements.Y_label, Y_label);
+    await user.click(elements.generate_chart_button);
+
+    // assert that our function was called
+    expect(gen_chart_spy).toHaveBeenCalled(); 
+    
+	// this function will refresh everything for our future tests! (+ get rid
+    // of spys middleware):)
+    await functions.resetForNextTest(elements);
+    gen_chart_spy.mockRestore();
+
+}
+
```

Comment:
Maybe add some more tests, especially to check when the spy is not called? This is also covered on the client side in the testDisplayAlerts, but it might be good to check that the request is not called when an invalid input is entered.

Reply:
attempted to do this but when applying the spy to the alert function and attempting to input valid inputs (X + Y label and one data point) i run into an error. it looks like elsewhere in the code, alert is called with the parameter "Error generating chart: ·ReferenceError: fetch is not defined". im guessing this is just some jsDOM specific stuff that is causing this? not sure i have the time to dig into it for the time being.